lib/stripe/subscriptions/coupon.ex

```diff
@@ -17,21 +17,23 @@ defmodule Stripe.Coupon do
   import Stripe.Request
 
   @type t :: %__MODULE__{
-          id: Stripe.id(),
-          object: String.t(),
-          amount_off: pos_integer,
-          created: Stripe.timestamp(),
-          currency: String.t(),
-          duration: String.t(),
-          duration_in_months: pos_integer | nil,
-          livemode: boolean,
-          max_redemptions: pos_integer,
-          metadata: Stripe.Types.metadata(),
-          percent_off: pos_integer,
-          redeem_by: Stripe.timestamp(),
-          times_redeemed: non_neg_integer,
-          valid: boolean
-        }
+               id: Stripe.id,
+               object: String.t,
+               amount_off: pos_integer | nil,
+               created: Stripe.timestamp(),
+               currency: String.t | nil,
+               duration: String.t,
+               duration_in_months: pos_integer | nil,
+               livemode: boolean,
+               max_redemptions: pos_integer | nil,
+               metadata: %{
+                 optional(String.t) => String.t
+               },
+               percent_off: pos_integer | nil,
+               redeem_by: Stripe.timestamp() | nil,
+               times_redeemed: non_neg_integer,
+               valid: boolean
```

Comment:
In elixir 1.6 all functions/types will be formatted with parentheses, so it will be `Stripe.id()`, `String.t()`, etc.

I'm not sure if that's final, but it's in RC, so going back on it here probably means more work later. Mind reverting to parentheses?